engine/client/gl_rmain.c

```diff
@@ -1326,15 +1327,89 @@ void R_RenderFrame( const ref_params_t *fd, qboolean drawWorld )
 	GL_BackendEndFrame();
 }
 
+inline static void gl_InsertBlackFrame( void )
+{
+	if (CL_IsInConsole()) // No strobing on the console
+	{
+		pglEnable(GL_SCISSOR_TEST);
+		pglScissor(con_rect.x, (-con_rect.y) - (con_rect.h*1.25), con_rect.w, con_rect.h);
+		pglClearColor(0.0f, 0.0f, 0.0f, 1.0f);
+		pglClear(GL_COLOR_BUFFER_BIT);
+		pglDisable(GL_SCISSOR_TEST);
+	}
+	else
+	{
+		pglClearColor(0.0f, 0.0f, 0.0f, 1.0f);
+		pglClear(GL_COLOR_BUFFER_BIT);
+	}
+}
+
+/*
+===============
+R_Strobe
+
+TODO: Consider vsync timings and do not render the supposed black frame at all.
+===============
+*/
+void R_Strobe( void )
+{
+	int getInterval = r_strobe->integer; // Check through modified tag first?
```

Comment:
You might want move ```sCounter``` to R_Strobe function. Just declare it static, and it will be saved between function calls.